### DISCUSSION TITLE
IR Diode

### DISCUSSION THREAD

**Discussion by jmino** [PlummersSoftwareLLC/NightDriverStrip#86] [General]
Hello there,

In one of the videos you described how to add an IR Diode to the ESP32 and use a remote to change the color of the lights. I can not seem to recall in which video you did just that. I was wondering if you could describe again how to add an IR Diode to the ESP32 or perhaps point me to the video when you do talk about adding an IR Diode.

Thanks again!

**Comment by Grimwig** [comment]
I don't think remote control has been covered on the YouTube channel but if you want to give it a go in the meantime, here's what I did:

1. Connect an IR sensor to a free IO pin and to ground & 5 volts. I bought a cheap kit which included the remote and sensor for £2 or so. Search the Internet for 'arduino ir receiver'. The sensors are either bare or on a breakout board. The breakout boards have an LED that flashes when the sensor receives data, which is useful when first getting things up and running.

2. Enable IR remote functionality by adding these lines your build section in platformio.ini
     -D_IR_ENABLE_DEFAULT_=false
     -DDECODE_NEC=true //most cheap Arduino-style pocket remotes have NEC encoding
3. Add these lines to the relevant section in globals.h. Change '35' to whichever IO PIN you connected to your IR sensor.
     #define IR_REMOTE_PIN 35
     #define ENABLE_REMOTE   1 

4. At this point, you should see some decoder values in the serial monitor window whenever you press a key on the remote. The process now is to match those values to the function you want each key to perform. This is done by editing remotecontrol.h. Here are some example lines from my config:

#define keyElegoo true

#if keyElegoo
#define	IR_BPLUS  0xFF906F	// UP 
#define	IR_BMINUS 0xFFE01F	// DOWN 
#define	IR_ON 	  0xFFA25D	// ON
#define	IR_OFF 	  0xFFE21D	// OFF
#define	IR_R 	  0xFF30CF	// 1
#define	IR_G 	  0xFF18E7	// 2 
#define	IR_B  	  0xFF7A85	// 3 
#define	IR_W 	  0xFF6897	// 0 

These hex codes are specific to my remote and the comments refer to the legend on each button. Your codes will most likely be different. I didn't really edit much else in remotecontrol.h apart from changing some of the RGB values in the section starting RemoteColorCodes[] =. I looked up the RGB values for various shades of white and assigned those to some other buttons. Hope this helps get you going.

**Reply by airnocker** [reply]
@Grimwig, your almost 2 year old posting for enabling an IR remote was enormously helpful to me, very clear and concise!  Thanks
